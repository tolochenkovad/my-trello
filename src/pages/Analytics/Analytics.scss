.analytics {
  padding: 20px 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 60vw;
  height: 60vh;
  margin: 0 auto;
}