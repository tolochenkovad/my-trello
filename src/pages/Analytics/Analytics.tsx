import React, { FC } from 'react';
import { Chart } from 'react-google-charts';
import { useSelector } from 'react-redux';
import { getQuantityItemsInCategories } from '../../store/Tasks/selectors';
import { Spinner } from 'react-bootstrap';

const Analytics: FC = () => {
  const dataForChart = useSelector(getQuantityItemsInCategories);

  if (dataForChart.slice(1).every((item) => item[1] == '0')) {
    return <div className="text-center">No data. Please create tasks to display analytics.</div>;
  }

  return (
    <div className="analytics">
      <h3 className="text-center">Tasks statistics by categories</h3>
      <Chart
        chartType="PieChart"
        options={{
          chartArea: {
            left: 0,
            right: 0,
            top: 20,
            bottom: 0,
          },
          legend: {
            position: 'right',
            alignment: 'center',
            textStyle: {
              fontSize: 14,
            },
          },
        }}
        loader={<Spinner animation="border" />}
        data={dataForChart}
        width="100%"
        height="100%"
      />
    </div>
  );
};

export default Analytics;
