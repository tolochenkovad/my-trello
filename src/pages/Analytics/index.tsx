import React, { FC, Suspense, lazy } from 'react';
import { useSelector } from 'react-redux';
import { getQuantityItemsInCategories } from '../../store/Tasks/selectors';
import { Spinner } from 'react-bootstrap';
import classes from './Analytics.module.scss';

const Chart = lazy(() => import('react-google-charts').then((module) => ({ default: module.Chart })));

const Analytics: FC = () => {
  const dataForChart = useSelector(getQuantityItemsInCategories);

  if (dataForChart.slice(1).every((item) => item[1] == '0')) {
    return <div className="text-center">No data. Please create tasks to display analytics.</div>;
  }

  return (
    <div className={classes.analytics}>
      <h3 className="text-center">Tasks statistics by categories</h3>
      <Suspense fallback={<Spinner animation="border" />}>
        <Chart
          chartType="PieChart"
          options={{
            chartArea: {
              left: 0,
              right: 0,
              top: 20,
              bottom: 0,
            },
            legend: {
              position: 'right',
              alignment: 'center',
              textStyle: {
                fontSize: 14,
              },
            },
          }}
          loader={<Spinner animation="border" />}
          data={dataForChart}
          width="100%"
          height="100%"
        />
      </Suspense>
    </div>
  );
};

export default Analytics;
