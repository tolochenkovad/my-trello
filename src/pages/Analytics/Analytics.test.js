import React from 'react';
import { shallow } from 'enzyme';
import Analytics from './Analytics';
import { useSelector } from 'react-redux';

let wrapper;

const dataForChart = [
  ['Task', 'Quantity'],
  ['To do', 2],
  ['In progress', 1],
  ['Done', 0],
];

useSelector.mockImplementation(() => dataForChart);

beforeEach(() => {
  wrapper = shallow(<Analytics />);
});

describe('Analytics', () => {
  it('should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('children is exist', () => {
    it('Title', () => {
      expect(wrapper.find('h3')).toHaveLength(1);
    });

    it('Chart', () => {
      expect(wrapper.find('lazy').props().chartType).toEqual('PieChart');
    });
  });
});
