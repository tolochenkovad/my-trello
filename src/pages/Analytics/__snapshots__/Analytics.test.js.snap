// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics should render correctly 1`] = `
<div
  className="analytics"
>
  <h3
    className="text-center"
  >
    Tasks statistics by categories
  </h3>
  <Suspense
    fallback={
      <Spinner
        animation="border"
      />
    }
  >
    <lazy
      chartType="PieChart"
      data={
        Array [
          Array [
            "Task",
            "Quantity",
          ],
          Array [
            "To do",
            2,
          ],
          Array [
            "In progress",
            1,
          ],
          Array [
            "Done",
            0,
          ],
        ]
      }
      height="100%"
      loader={
        <Spinner
          animation="border"
        />
      }
      options={
        Object {
          "chartArea": Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
            "top": 20,
          },
          "legend": Object {
            "alignment": "center",
            "position": "right",
            "textStyle": Object {
              "fontSize": 14,
            },
          },
        }
      }
      width="100%"
    />
  </Suspense>
</div>
`;
